content/blog/dplyr-1-0-4-if-any/index.Rmd

```diff
@@ -0,0 +1,111 @@
+    Minor release with major performance improvements for `across()` and 
+    two new functions `if_any()` and `if_all()`.
+
+photo:
+  url: https://unsplash.com/photos/nQz49efZEFs
+  author: Mattias Olsson
+
+categories: [package] 
+tags: [dplyr]
+---
+
+We're happy to announce the release of [dplyr](https://dplyr.tidyverse.org) 1.0.4, 
+discussed in this blog post. The highlights of this release are: two new functions `if_all()` and `if_any()`, and improved performance improvements of `across()`. 
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dplyr")
+```
+
+You can see a full list of changes in the [release notes](https://github.com/tidyverse/dplyr/releases/tag/v1.0.4). 
+
+```{r setup, echo=FALSE}
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## if_any() and if_all()
+
+The new `across()` function introduced as part of 
+[dplyr 1.0.0](https://www.tidyverse.org/blog/2020/04/dplyr-1-0-0-colwise/) is proving to be 
+a successful addition to dplyr. In case you missed it, `across()` lets you 
+conveniently express a set of actions to be performed across a tidy selection 
+of columns. 
+
+`across()` is very useful within `summarise()` and `mutate()`, but it can be confusing when used in `filter()` because it is not clear how the results can be combined into one logical vector. 
+So to fill the gap, we're introducing two new functions `if_all()` and `if_any()`. Let's 
+directly dive in to an example: 
+
+```{r}
+library(palmerpenguins)
+
+big <- function(x) {
+  x > mean(x, na.rm = TRUE)
+}
+
+# keep rows if all the selected columns are "big"
+penguins %>% 
+  filter(if_all(contains("bill"), big))
+
+# keep rows where at least one of the columns is "big"
+penguins %>% 
+  filter(if_any(contains("bill"), big))
+```
+
+Both functions operate similarly to `across()` but go the extra mile of aggregating the 
+results to indicate if *all* the results are true when using `if_all()`, or if *at least 
+one* is true when using `if_any()`. 
+
+Although `if_all()` and `if_any()` were designed with `filter()` in mind, we 
+[then discovered](https://github.com/tidyverse/dplyr/issues/5709) that they can also be
+useful within `mutate()` and/or `summarise()`. 
+
+```{r}
+penguins %>% 
+  filter(!is.na(bill_length_mm)) %>% 
+  mutate(
+    category = case_when(
+      if_all(contains("bill"), big) ~ "both big", 
+      if_any(contains("bill"), big) ~ "one big", 
+      TRUE                          ~ "small"
+    )) %>% 
+  count(category)
+```
+
+## Faster across()
+
+One of the main motivations for across() was eliminating the need for 
+every verb to have a `_at`, `_if`, and `_all` variant. Unfortunately, however, this came with a performance cost. In this release, we have redesigned `across()` to eliminate that problem when possible. 
+
+```{r}
+library(vroom)
+
+mun2014 <- vroom("https://data.regardscitoyens.org/elections/2014_municipales/MN14_Bvot_T1_01-49.txt", 
+                 col_select = -c('X4','X9','X10','X11'),
```

Comment:
Add `col_types = list()` to suppress the type output

Reply:
Thanks